### DISCUSSION TITLE
Example cannot work

### DISCUSSION THREAD

**Discussion by cathaysia** [eProsima/Fast-DDS#5145] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

DDSHelloWorldExample cannot work.

### Current behavior

after v2.13.0, the examples cannot work. DDSHelloWorldExample cannot work.

### Steps to reproduce

```console
cmake -B build -GNinja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="~/FastDDS/install;~/FastDDS/foonathan_memory_vendor/build/foo_mem_ext_prj_install" -DEPROSIMA_BUILD=ON -DSQLITE3_SUPPORT=OFF -DSHM_TRANSPORT_DEFAULT=OFF -DCOMPILE_TOOLS=OFF -DCMAKE_TOOLCHAIN_FILE=$HOME/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DSECURITY=OFF -DCOMPILE_EXAMPLES=ON -DSHM_TRANSPORT_DEFAULT=OFF
cmake --build build
# run DDSHelloWorldExample
````

### Fast DDS version/commit

v2.14.3

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

macos 14.6

**Comment by cathaysia** [comment]
I have not do a git bitsect, only know 2.13.0 can work.

**Comment by JesusPoderoso** [comment]
Hi @cathaysia, thanks for using Fast DDS.
We would need more information on the context, tracebacks or logging info. 
The [currently supported MacOS platform](https://fast-dds.docs.eprosima.com/en/stable/notes/versions.html#build-system-dependencies) is Mojave 10.14, with CLang 15 compiler (which [builds the example successfully](https://github.com/eProsima/Fast-DDS/actions/runs/10276555000/job/28437138134#step:13:4014)).
For that reason, I am moving this ticket to the support section according to the [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).

**Comment by conroy-cheers** [comment]
Just ran a bisect on this; the culprit is 7832901d34c5fff09609411ce0d4b5b124919e76

https://github.com/eProsima/Fast-DDS/blob/7832901d34c5fff09609411ce0d4b5b124919e76/src/cpp/utils/threading/threading_osx.ipp#L136-L143

Commenting out the above `EPROSIMA_LOG_ERROR` call makes the pub/sub example work again. I added some extra checks and it appears setting the thread affinity fails both before and after the problem commit (macOS 15.1 Sequoia) - perhaps trying to log at that point is causing something to get stuck?

As a crappy workaround, setting the `LOG_NO_ERROR` CMake flag seems to get things working again.

**Comment by Lovely-XPP** [comment]
It seems that Function `thread_policy_set` is not supported for M series chips (Return KERN_NOT_SUPPORTED = 46).

https://forums.developer.apple.com/forums/thread/699925

**Comment by cathaysia** [comment]
Hi everyone, I am no longer working on DDS. So I will close this issue. Anyone who has interesting on this problem can open an new issue.
